src/SignalR/clients/java/signalr/test/src/main/java/com/microsoft/signalr/TestLogger.java

```diff
@@ -27,7 +27,9 @@ public TestLogger(String category) {
     }
 
     public ILoggingEvent assertLog(String logMessage) {
-        for (ILoggingEvent log : appender.list) {
+        // Copy items just in case logs are written while iterating
+        ILoggingEvent[] list = appender.list.toArray(new ILoggingEvent[1]);
```

Comment:
Couldn't toArray itself throw a `ConcurrentModificationException` since it also iterates over the list?

Reply:
Technically, although I think because of the list impl it wont ever.

But, I changed it anyways to just stop the logging on first call to `assertLog`